### PR TITLE
Update typing in Preact 8

### PR DESCRIPTION
**Pull request by 38elements**
This updates typing in Preact 8.
Since changing event handler is breaking change, I did not update it.

### CODE REVIEW

File: src/preact.d.ts [hunk 1]
```diff
@@ -442,8 +442,9 @@ declare namespace preact {
 		interface DOMAttributes extends preact.PreactDOMAttributes {
 			// Image Events
 			onLoad?: GenericEventHandler;
-			onError?: GenericEventHandler;
 			onLoadCapture?: GenericEventHandler;
+			onError?: GenericEventHandler;
+			onErrorCapture?: GenericEventHandler;
```
Comment (38elements) [inline src/preact.d.ts]: https://github.com/preactjs/preact/commit/9a982bdec8596fff13f66fde503780e0b697abe8#diff-70083611527b9d7d728e1e14dd8f977f

File: src/preact.d.ts [hunk 2]
```diff
@@ -1225,8 +1247,9 @@ declare global {
...
```
Comment (38elements) [inline src/preact.d.ts]: https://github.com/preactjs/preact/commit/9a982bdec8596fff13f66fde503780e0b697abe8#diff-70083611527b9d7d728e1e14dd8f977f

File: src/preact.d.ts [hunk 3]
```diff
@@ -477,6 +481,7 @@ declare namespace preact {
 			onSubmit?: GenericEventHandler;
 			onSubmitCapture?: GenericEventHandler;
 			onInvalid?: GenericEventHandler;
+			onInvalidCapture?: GenericEventHandler;
```
Comment (38elements) [inline src/preact.d.ts]: https://github.com/preactjs/preact/commit/7af0f62618bc28c8862bc4462ac4fbd764a786e1#diff-70083611527b9d7d728e1e14dd8f977f

File: src/preact.d.ts [hunk 4]
```diff
@@ -1260,6 +1286,7 @@ declare global {
...
```
Comment (38elements) [inline src/preact.d.ts]: https://github.com/preactjs/preact/commit/7af0f62618bc28c8862bc4462ac4fbd764a786e1#diff-70083611527b9d7d728e1e14dd8f977f

File: src/preact.d.ts [hunk 5]
```diff
@@ -1596,6 +1639,7 @@ declare global {
 			h6: HTMLAttributes;
 			head: HTMLAttributes;
 			header: HTMLAttributes;
+			hgroup: HTMLAttributes;
```
Comment (38elements) [inline src/preact.d.ts]: https://github.com/preactjs/preact/pull/1340/files

File: src/preact.d.ts [hunk 6]
```diff
@@ -813,6 +834,7 @@ declare namespace preact {
...
```
Comment (38elements) [inline src/preact.d.ts]: https://github.com/preactjs/preact/pull/1340/files

File: src/preact.d.ts [hunk 7]
```diff
@@ -636,9 +641,14 @@ declare namespace preact {
 			allowFullScreen?: boolean;
 			allowTransparency?: boolean;
 			alt?: string;
+			as?: string;
 			async?: boolean;
 			autocomplete?: string;
+			autoComplete?: string;
```
Comment (38elements) [inline src/preact.d.ts]: https://github.com/preactjs/preact/pull/1390/files

File: src/preact.d.ts [hunk 8]
```diff
@@ -636,9 +641,14 @@ declare namespace preact {
...
+			autocorrect?: string;
+			autoCorrect?: string;
 			autofocus?: boolean;
+			autoFocus?: boolean;
```
Comment (38elements) [inline src/preact.d.ts]: https://github.com/preactjs/preact/pull/1390/files

File: src/preact.d.ts [hunk 9]
```diff
@@ -680,6 +691,7 @@ declare namespace preact {
 			href?: string;
 			hrefLang?: string;
 			for?: string;
+			htmlFor?: string;
```
Comment (38elements) [inline src/preact.d.ts]: https://github.com/preactjs/preact/pull/1390/files

File: src/preact.d.ts [hunk 10]
```diff
@@ -1419,9 +1446,14 @@ declare global {
 			allowFullScreen?: boolean;
 			allowTransparency?: boolean;
 			alt?: string;
+			as?: string;
 			async?: boolean;
 			autocomplete?: string;
+			autoComplete?: string;
```
Comment (38elements) [inline src/preact.d.ts]: https://github.com/preactjs/preact/pull/1390/files

File: src/preact.d.ts [hunk 11]
```diff
@@ -1419,9 +1446,14 @@ declare global {
...
+			autocorrect?: string;
+			autoCorrect?: string;
 			autofocus?: boolean;
+			autoFocus?: boolean;
```
Comment (38elements) [inline src/preact.d.ts]: https://github.com/preactjs/preact/pull/1390/files

File: src/preact.d.ts [hunk 12]
```diff
@@ -1463,6 +1496,7 @@ declare global {
 			href?: string;
 			hrefLang?: string;
 			for?: string;
+			htmlFor?: string;
```
Comment (38elements) [inline src/preact.d.ts]: https://github.com/preactjs/preact/pull/1390/files

File: src/preact.d.ts [hunk 13]
```diff
@@ -1419,9 +1446,14 @@ declare global {
 			allowFullScreen?: boolean;
 			allowTransparency?: boolean;
 			alt?: string;
+			as?: string;
 			async?: boolean;
 			autocomplete?: string;
+			autoComplete?: string;
+			autocorrect?: string;
+			autoCorrect?: string;
```
Comment (38elements) [inline src/preact.d.ts]: https://github.com/preactjs/preact/pull/1393/files

File: src/preact.d.ts [hunk 14]
```diff
@@ -636,9 +641,14 @@ declare namespace preact {
...
```
Comment (38elements) [inline src/preact.d.ts]: https://github.com/preactjs/preact/pull/1393/files

File: src/preact.d.ts [hunk 15]
```diff
@@ -762,6 +776,13 @@ declare namespace preact {
 			resource?: string;
 			typeof?: string;
 			vocab?: string;
+
+			// Microdata Attributes
+			itemProp?: string;
+			itemScope?: boolean;
+			itemType?: string;
+			itemID?: string;
+			itemRef?: string;
```
Comment (38elements) [inline src/preact.d.ts]: https://github.com/preactjs/preact/pull/1585/files

File: src/preact.d.ts [hunk 16]
```diff
@@ -1545,6 +1581,13 @@ declare global {
...
```
Comment (38elements) [inline src/preact.d.ts]: https://github.com/preactjs/preact/pull/1585/files

File: src/preact.d.ts [hunk 17]
```diff
@@ -1532,6 +1567,7 @@ declare global {
 			type?: string;
 			useMap?: string;
 			value?: string | string[] | number;
+			volume?: string | number;
```
Comment (38elements) [inline src/preact.d.ts]: https://github.com/preactjs/preact/pull/1938/files

File: src/preact.d.ts [hunk 18]
```diff
@@ -749,6 +762,7 @@ declare namespace preact {
...
```
Comment (38elements) [inline src/preact.d.ts]: https://github.com/preactjs/preact/pull/1938/files

File: src/preact.d.ts [hunk 19]
```diff
@@ -661,8 +671,9 @@ declare namespace preact {
 			dateTime?: string;
 			default?: boolean;
 			defer?: boolean;
-			dir?: string;
+			dir?: 'auto' | 'rtl' | 'ltr';
 			disabled?: boolean;
+			disableRemotePlayback?: boolean;
```
Comment (38elements) [inline src/preact.d.ts]: https://github.com/preactjs/preact/pull/1955/files

File: src/preact.d.ts [hunk 20]
```diff
@@ -1444,8 +1476,9 @@ declare global {
...
```
Comment (38elements) [inline src/preact.d.ts]: https://github.com/preactjs/preact/pull/1955/files

File: src/preact.d.ts [hunk 21]
```diff
@@ -707,6 +719,7 @@ declare namespace preact {
 			multiple?: boolean;
 			muted?: boolean;
 			name?: string;
+			nonce?: string;
```
Comment (38elements) [inline src/preact.d.ts]: https://github.com/preactjs/preact/commit/2a3f92d7d6f5a046f2ba28ced845048d9e94c1e7#diff-70083611527b9d7d728e1e14dd8f977f

File: src/preact.d.ts [hunk 22]
```diff
@@ -1490,6 +1524,7 @@ declare global {
...
```
Comment (38elements) [inline src/preact.d.ts]: https://github.com/preactjs/preact/commit/2a3f92d7d6f5a046f2ba28ced845048d9e94c1e7#diff-70083611527b9d7d728e1e14dd8f977f

File: src/preact.d.ts [hunk 23]
```diff
@@ -1419,9 +1446,14 @@ declare global {
 			allowFullScreen?: boolean;
 			allowTransparency?: boolean;
 			alt?: string;
+			as?: string;
```
Comment (38elements) [inline src/preact.d.ts]: https://github.com/preactjs/preact/pull/2068/files

File: src/preact.d.ts [hunk 24]
```diff
@@ -636,9 +641,14 @@ declare namespace preact {
...
```
Comment (38elements) [inline src/preact.d.ts]: https://github.com/preactjs/preact/pull/2068/files

File: src/preact.d.ts [hunk 25]
```diff
@@ -661,8 +671,9 @@ declare namespace preact {
 			dateTime?: string;
 			default?: boolean;
 			defer?: boolean;
-			dir?: string;
+			dir?: 'auto' | 'rtl' | 'ltr';
```
Comment (38elements) [inline src/preact.d.ts]: https://github.com/preactjs/preact/pull/2108/files

File: src/preact.d.ts [hunk 26]
```diff
@@ -1444,8 +1476,9 @@ declare global {
...
```
Comment (38elements) [inline src/preact.d.ts]: https://github.com/preactjs/preact/pull/2108/files

File: src/preact.d.ts [hunk 27]
```diff
@@ -1244,6 +1267,9 @@ declare global {
 			onCompositionUpdate?: CompositionEventHandler;
 			onCompositionUpdateCapture?: CompositionEventHandler;
 
+			// Details Events
+			onToggle?: GenericEventHandler;
```
Comment (38elements) [inline src/preact.d.ts]: https://github.com/preactjs/preact/pull/2151/files

File: src/preact.d.ts [hunk 28]
```diff
@@ -461,6 +462,9 @@ declare namespace preact {
...
```
Comment (38elements) [inline src/preact.d.ts]: https://github.com/preactjs/preact/pull/2151/files

**Comment by developit** [comment]
wow, this looks like it was a ton of work. Thank you!!

**Pull request merged**